### DISCUSSION TITLE
Chen-Gackstatter surface

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7827] [Show and tell]
Hello,

I propose a game. Could you guess the shape of the ARAP parameterization space for the Chen-Gackstatter surface?

![ChenGackstatter](https://github.com/CGAL/cgal/assets/4466543/79aefe4d-f401-4c16-97c1-4571ecbb882f)

**Comment by stla** [comment]
Argh, we'll never know:
```
Error: Cannot solve linear system
```

**Comment by stla** [comment]
DCP perfectly gets the shape of the parameterization space:

![UV-ChenGackstatter-DCP](https://github.com/CGAL/cgal/assets/4466543/4da22611-a9b9-4eb1-b528-241ef33cadd2)

**Comment by stla** [comment]
Hmm but it is not usable to map a radial checkerboard. This surface is strange.

![ChenGackstatterCheckerboard](https://github.com/CGAL/cgal/assets/4466543/565501d0-4204-4d16-a077-0ca88270b74f)

**Comment by stla** [comment]
CGAL finds two borders of this mesh. However we see only one, no?

**Comment by sloriot** [comment]
To be honest, on the animation I see the hole moving while rotating and I don't understand what I see. About the two borders maybe the mesh contains more than what you expect.

**Reply by stla** [reply]
This mesh is obtained with a parametrization (r, t) with r in (0.2, 0.8) and t in (0, 2pi), and we use z=r*e^(i*theta) in the formulas. I tried to change the bounds of (0.2, 0.8) and this gives very weird results. If I decrease 0.8 I see another border.

**Comment by stla** [comment]
Here are the two borders found by CGAL. Really strange.

![ChenGackstatter-borders](https://github.com/CGAL/cgal/assets/4466543/464ef2f1-e953-4cc9-9bbf-20e889f4a7f3)

**Comment by sloriot** [comment]
Can you share the mesh?

**Reply by stla** [reply]
Here is a non-binary PLY file of this mesh.
[ChenGackstatter.ply.txt](https://github.com/CGAL/cgal/files/13166785/ChenGackstatter.ply.txt)

**Comment by sloriot** [comment]
The mesh is self-interesting. It is folding on itself.

**Comment by stla** [comment]
I can get a non-self-intersecting version by alpha-wrapping the vertices 👍

**Reply by stla** [reply]
Ah but this new mesh does not have a border :-(
